bazel/sh_test_wrapper.sh

```diff
@@ -3,4 +3,6 @@
 # Where the runfiles are for tests.
 export TEST_RUNDIR="${TEST_SRCDIR}/${TEST_WORKSPACE}"
 
+cd $(dirname "$0")
```

Comment:
I can rewrite the cd command as `cd "$(dirname "${TEST_RUNDIR}"/${TEST_BINARY}")"`.

Ultimately, the files specified in srcs end up in `"${TEST_SRCDIR}"/"${TEST_WORKSPACE}"/path/to/file` (see https://docs.bazel.build/versions/master/build-ref.html#data). The script sets `TEST_RUNDIR` to `"${TEST_SRCDIR}"/"${TEST_WORKSPACE}"`. The path/to/file here is test/integration. But envoy_sh_test is also used for //test/tools/router_check/test:router_tool_test, which will use a different path.

I've no doubt bazel is capable of doing what you want, I just don't understand how to do it. I've spent quite a bit of time reading Bazel docs. The only way to access the repository context, that I know of, is to write a custom rule, so I guess I'll try that again.